### PR TITLE
Docker and docker-compose support

### LINKED ISSUE #22: use docker to run CI
Instead of running jobs directly, you should use docker.

e.g. https://github.com/spotify/docker-client

### PR DESCRIPTION
**Pull request by kounoike**
In this PR, I add basic docker & docker-compose support to docker-ci plugin. It can close #22.

- Administrator can choose enable/disable docker/docker-compose
- also can specify docker (or docker-compose) command. (e.g. Ubuntu with snap: /snap/bin/docker)
- user can specify Dockerfie/docker-compose.yml path
    - currently, docker-compose servicename is restricted to "ci"

I only tested this PR on Windows 10 Pro.

### CODE REVIEW

**Comment by kounoike** [comment]
### docker example





### docker-compose example

File: src/main/scala/io/github/gitbucket/ci/manager/BuildJobThread.scala [hunk 1]
```diff
@@ -213,6 +215,92 @@ class BuildJobThread(queue: LinkedBlockingQueue[BuildJob], threads: LinkedBlocki
...
+    val runContainerCommand = s"${dockerCommand} run --rm --name ${containerName} ${tagName}"
+
+    sb.append(s"${buildContainerCommand}\n")
+    val buildResult = runProcess(job, buildDir, workspaceDir, buildContainerCommand)
+    if (buildResult == 0){
+      sb.append(s"${runContainerCommand}\n")
+      val exitCode = runProcess(job, buildDir, workspaceDir, runContainerCommand)
+
+      val imageId = Process(s"""${dockerCommand} images --format "{{.ID}}" ${tagName}""").!!.stripLineEnd
+      println(imageId)
```
Comment (takezoe) [inline src/main/scala/io/github/gitbucket/ci/manager/BuildJobThread.scala]: Is this `println` statement for debug?

File: src/main/twirl/gitbucket/ci/config.scala.html [hunk 2]
```diff
@@ -17,7 +18,7 @@
               </label>
             </fieldset>
             <div id="buildSettings">
-              <fielset class="form-group">
+              <fieldset class="form-group">
```
Comment (takezoe) [inline src/main/twirl/gitbucket/ci/config.scala.html]: 👍

File: src/main/scala/io/github/gitbucket/ci/manager/BuildJobThread.scala (hunk 1 above)
Comment (kounoike) [reply src/main/scala/io/github/gitbucket/ci/manager/BuildJobThread.scala]: Sorry!

**Comment by kounoike** [comment]
@takezoe I fixed your suggestion. And fixed system-wide enableDockerCompose option.

**Comment by takezoe** [comment]
@kounoike Thanks!

**Pull request merged**